src/library/scala/sys/process/Process.scala

```diff
@@ -172,6 +174,50 @@ trait ProcessCreation {
     require(files.nonEmpty)
     files.map(_.cat).reduceLeft(_ #&& _)
   }
+
+  /**
+    * Echo the given text.
+    *
+    * This can be used as the source of input for a process, for example:
+    *
+    * {{{
+    * import scala.sys.process._
+    * import scala.sys.process.Process.echo
+    *
+    * echo("Hello, world!") #> "curl -d - https://example.com/upload" !
+    * }}}
+    *
+    * Unlike the shell echo command, this does not append a newline to the text.
+    *
+    * @param text The text to echo.
+    */
+  def echo(text: String): Source = {
+    new ProcessBuilder.IStreamBuilder(
+      new ByteArrayInputStream(text.getBytes()),
```

Comment:
Is using the platform encoding correct here, or should it take an encoding as an argument?

Reply:
I think platform encoding is correct. I did consider taking an implicit for the encoding, but the process API already assumes platform encoding everywhere else that it handles text, eg all the output handling (eg, `.!!`) uses the platform encoding, and in shell scripting in general, platform encoding is assumed, eg by all the unix commands that handle text (grep, sed, etc).